backend/src/main/java/org/eclipse/tractusx/puris/backend/controller/MaterialPartnerRelationsController.java

```diff
@@ -0,0 +1,148 @@
+        @ApiResponse(responseCode = "500", description = "Internal Server Error")
+    })
+    public ResponseEntity<?> createMaterialPartnerRelation(
+        @Parameter(name = "ownMaterialNumber", description = "The Material Number that is used in your own company to identify the Material.",
+            example = "MNR-7307-AU340474.002", required = true) @RequestParam String ownMaterialNumber,
+        @Parameter(name = "partnerBpnl", description = "The unique BPNL that was assigned to that Partner.",
+            example = "BPNL2222222222RR", required = true) @RequestParam() String partnerBpnl,
+        @Parameter(name = "partnerMaterialNumber", description = "The Material Number that this Partner is using in his own company to identify the Material.",
+            example = "MNR-8101-ID146955.001") @RequestParam String partnerMaterialNumber,
+        @Parameter(name = "partnerSupplies", description = "This boolean flag indicates whether this Partner is a potential supplier of the given Material.",
+            example = "true") @RequestParam boolean partnerSupplies,
+        @Parameter(name = "partnerBuys", description = "This boolean flag indicates whether this Partner is a potential customer of this Material.",
+            example = "true") @RequestParam boolean partnerBuys) {
+        Material material = materialService.findByOwnMaterialNumber(ownMaterialNumber);
+        if (material == null || partnerBpnl == null) {
+            return new ResponseEntity<>(HttpStatusCode.valueOf(400));
+        }
+        Partner partner = null;
+        if (partnerBpnl != null) {
+            partner = partnerService.findByBpnl(partnerBpnl);
+        }
+        if (partner == null) {
+            return new ResponseEntity<>(HttpStatusCode.valueOf(400));
+        }
+
+        if (mprService.find(material, partner) != null) {
+            return new ResponseEntity<>(HttpStatusCode.valueOf(409));
+        }
+        MaterialPartnerRelation newMpr = new MaterialPartnerRelation(material, partner, partnerMaterialNumber, partnerSupplies, partnerBuys);
+
+        newMpr = mprService.create(newMpr);
+        if (newMpr == null) {
+            return new ResponseEntity<>(HttpStatusCode.valueOf(500));
+        }
+
+        return new ResponseEntity<>(HttpStatusCode.valueOf(200));
+    }
+
+    @PutMapping
+    @CrossOrigin
+    @Operation(description = "Updates an existing MaterialPartnerRelation. You have to specify the ownMaterialNumber and " +
+        "the partnerBpnl. The other three parameters are genuinely optional. Provide them only if you want to change their values. ")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "Update was accepted."),
+        @ApiResponse(responseCode = "404", description = "No existing entity was found. "),
+        @ApiResponse(responseCode = "500", description = "Internal Server Error")
+    })
+    public ResponseEntity<?> updateMaterialPartnerRelation(
```

Comment:
same for ownMaterialNumber and partnerBpnl

Reply:
See above